keps/sig-network/2004-topology-aware-subsetting/README.md

```diff
@@ -565,6 +583,165 @@ label. Each approach may group the endpoints in slightly different
+  If the proportion of `endpoint_slice_controller/syncs` with a "failure" result
+  is greater than 10%, a rollback may be considered. It is worth noting that
+  other issues can cause sync failures such as an out of date informer cache.
+  The key indicator should be a significantly elevated error rate when compared
+  with before the feature was enabled.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  This feature is not yet implemented but per-Service enablement/disablement is
+  covered in depth as part of the test plan.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+  Yes, this represents a replacement to the approach tracked with KEP 536. This
+  KEP included an alpha implementation but did not graduate beyond that.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  If the `endpointslices_changed_per_sync` metric has a non-zero value for the
+  `auto` approach, this feature is in use.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [x] Metrics
+    - Metric name: `endpoint_slice_controller/syncs`
+    - [Optional] Aggregation method: Counter
+    - Components exposing the metric: EndpointSlice Controller
+    - The relative failure rate over time can be used to track the health of
+      this controller.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  As a starting point, it is likely reasonable for the EndpointSlice controller
+  to experience up to a 10% sync failure rate. This is largely related to it
+  trying to update stale EndpointSlices. When we are able to find a solution for
+  that issue the expected sync failure rate should be significantly lower. This
+  specific problem is most notable for large Services that have rapidly updating
+  endpoints.
+
+* **Are there any missing metrics that would be useful to have to improve
+  observability of this feature?**
+  None that I can think of.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  This primarily depends on API Server. Kube-Proxy depends on this controller.
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+  provider?**
+  No.
+
+* **Will enabling / using this feature result in increasing size or count of the
+  existing API objects?**
+  This could result in as much as 3x the number of EndpointSlices depending on
+  the size of Services in a cluster. On average it will likely result in closer
+  to 50% more EndpointSlices depending on how widely this feature is used.
```

Comment:
Once it is default, it will converge on something between 1x and 3x, depending on the distribution of size of Services.  Worst case is 3x the number of slices, but updated 1/3 as often, right?

Reply:
Yeah, I think that's accurate. I tried to expand on this a bit to provide a more detailed representation of the number of EndpointSlices this would result in.